tokio/src/io/ready.rs

```diff
@@ -230,6 +254,9 @@ impl Ready {
             ready |= Ready::READ_CLOSED;
         }
 
+        // error readiness is reported regardless of interest
+        ready |= Ready::ERROR;
+
```

Comment:
Hmm. Does this mean that the `fn ready` methods on socket types always return `Error` unconditionally even if there are no `Ready` events?

We should probably have a test that `Ready` is not set if there have been no error events.

Reply:
the phrasing is misleading I think, I'll improve it. What is happening here is that with this change, an event with the error bit set (i.e. with the error readiness) matches any interest.

so no matter what interest a user configures, an event with error readiness is not discarded. That means the corresponding waker is woken up, and then can deal with the event as they see fit.

We already have some tests that perform an equality check on `Ready` values, e.g. 

```
#[tokio::test]
async fn clear_ready_matching_clears_ready() {
    use tokio::io::{Interest, Ready};

    let (a, mut b) = socketpair();

    let afd_a = AsyncFd::new(a).unwrap();
    b.write_all(b"0").unwrap();

    let mut guard = afd_a
        .ready(Interest::READABLE | Interest::WRITABLE)
        .await
        .unwrap();

    // the readiness is just readable and writable, not error 
    assert_eq!(guard.ready(), Ready::READABLE | Ready::WRITABLE);

    guard.clear_ready_matching(Ready::READABLE);
    assert_eq!(guard.ready(), Ready::WRITABLE);

    guard.clear_ready_matching(Ready::WRITABLE);
    assert_eq!(guard.ready(), Ready::EMPTY);
}
```

This test passes, so the error readiness is not set in this case.